From 1c4d5eea6ca3d5914b25613af2282ba6549325a6 Mon Sep 17 00:00:00 2001
From: Ivan Vecera <ivecera@redhat.com>
Date: Wed, 19 Nov 2014 16:14:45 -0500
Subject: [ethernet] r8169: call "rtl8168_driver_start" "rtl8168_driver_stop" only when hardware dash function is enabled

Message-id: <1416413687-12219-15-git-send-email-ivecera@redhat.com>
Patchwork-id: 100362
O-Subject: [RHEL7.1 PATCH 14/16] r8169:call "rtl8168_driver_start" "rtl8168_driver_stop" only when hardware dash function is enabled
Bugzilla: 1165764
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

BZ: #1165764

Upstream commit(s):
commit ee7a1beb9759c94aea67dd887faf5e447a5c6710
Author: Chun-Hao Lin <hau@realtek.com>
Date:   Wed Oct 1 23:17:21 2014 +0800

    r8169:call "rtl8168_driver_start" "rtl8168_driver_stop" only when hardware dash function is enabled

    These two functions are used to inform dash firmware that driver is been
    brought up or brought down. So call these two functions only when hardware dash
    function is enabled.

    Signed-off-by: Chun-Hao Lin <hau@realtek.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Ivan Vecera <ivecera@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/realtek/r8169.c b/drivers/net/ethernet/realtek/r8169.c
index 5afdc63..a8c51a7 100644
--- a/drivers/net/ethernet/realtek/r8169.c
+++ b/drivers/net/ethernet/realtek/r8169.c
@@ -7387,9 +7387,10 @@ static void rtl_remove_one(struct pci_dev *pdev)
  struct net_device *dev = pci_get_drvdata(pdev);
  struct rtl8169_private *tp = netdev_priv(dev);
 
- if (tp->mac_version == RTL_GIGA_MAC_VER_27 ||
-     tp->mac_version == RTL_GIGA_MAC_VER_28 ||
-     tp->mac_version == RTL_GIGA_MAC_VER_31) {
+ if ((tp->mac_version == RTL_GIGA_MAC_VER_27 ||
+      tp->mac_version == RTL_GIGA_MAC_VER_28 ||
+      tp->mac_version == RTL_GIGA_MAC_VER_31) &&
+     r8168_check_dash(tp)) {
   rtl8168_driver_stop(tp);
  }
 
@@ -7822,9 +7823,10 @@ static int rtl_init_one(struct pci_dev *pdev, const struct pci_device_id *ent)
       rtl_chip_infos[chipset].jumbo_tx_csum ? "ok" : "ko");
  }
 
- if (tp->mac_version == RTL_GIGA_MAC_VER_27 ||
-     tp->mac_version == RTL_GIGA_MAC_VER_28 ||
-     tp->mac_version == RTL_GIGA_MAC_VER_31) {
+ if ((tp->mac_version == RTL_GIGA_MAC_VER_27 ||
+      tp->mac_version == RTL_GIGA_MAC_VER_28 ||
+      tp->mac_version == RTL_GIGA_MAC_VER_31) &&
+     r8168_check_dash(tp)) {
   rtl8168_driver_start(tp);
  }
 
-- 
1.7.1